### PR TITLE
[SPARK-370][SPARK-499] Change image to run as nobody

### PR DESCRIPTION
**Pull request by ArtRand**
## Problem
The Spark image that is used for running the Dispatcher, Drivers, and Executors runs as `root` meaning that Marathon requires permissions to launch containers as `root` when on a strict-mode cluster. 

## Solution
Change default to `nobody`, allow access to directories in the image necessary for running Spark and it's components (e.g. nginx).

## Tested
CI unit tests for basic functionality, manually on a strict-mode cluster (see [JIRA](https://jira.mesosphere.com/browse/SPARK-370) for details) and manually against a Kerberized HDFS cluster

### Manual tests
- [x] Kerberos
- [x] History server

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -72,15 +72,18 @@ RUN ln -s /bin/grep /usr/bin/grep
 
 RUN ln -s /var/lib/runit/service/spark /etc/service/spark
 RUN ln -s /var/lib/runit/service/nginx /etc/service/nginx
+RUN mkdir -p /etc/hadoop
```
Comment (susanxhuynh) [inline docker/Dockerfile]: This is repeated in L. 60 above, one of them can be removed.

File: docker/Dockerfile [hunk 2]
```diff
@@ -76,11 +76,13 @@ RUN ln -s /var/lib/runit/service/nginx /etc/service/nginx
 ADD dist /opt/spark/dist
 
 # Commenting these for now, because we're running Spark as root in strict mode
```
Comment (benclarkwood) [inline docker/Dockerfile]: What does this comment mean?

Comment (benclarkwood) [reply docker/Dockerfile]: It seems like it should go away?

**Comment by skonto** [comment]
I am not so sure about nobody in general its a relic: https://lwn.net/Articles/695526/

**Pull request merged**